codec-http/src/main/java/io/netty/handler/codec/http/CookieUtil.java

```diff
@@ -90,7 +137,25 @@ static void add(StringBuilder sb, String name, long val) {
         sb.append((char) HttpConstants.SP);
     }
 
-    private CookieEncoderUtil() {
+    public static Character firstInvalidCookieNameOctet(String s) {
+        return firstInvalidOctet(s, VALID_COOKIE_NAME_OCTETS);
+    }
+
+    public static Character firstInvalidCookieValueOctet(String s) {
+        return firstInvalidOctet(s, VALID_COOKIE_VALUE_OCTETS);
+    }
+
+    private static Character firstInvalidOctet(String s, BitSet bits) {
```

Comment:
Could we just return the index integer (-1 when invalid octet is not found), to avoid the instantiation of `Character`?